lib/src/kbasesearchengine/tools/SearchTools.java

```diff
@@ -664,6 +679,12 @@ private void printError(final String msg) {
                 "Start an indexer worker with the provided id. At any given time, all workers " +
                 "MUST have unique ids. Set the id to '-' to generate a random id.")
         private String startWorker;
+
+        @Parameter(names = {"-f", "--reset-failed-events"}, description =
```

Comment:
I added todo comment in code for now. But I feel it may be simpler to just let some failed events to fail again. Once we know that all remaining failed events are pointless to run, we don't re-run them.

Reply:
But that means every time you have 1 new failed event you want to rerun, every other event that has ever failed will run first.